core/vm/memory.go

```diff
@@ -52,11 +54,12 @@ func (m *Memory) Set(offset, size uint64, value []byte) {
 func (m *Memory) Set32(offset uint64, val *uint256.Int) {
 	// length of store may never be less than offset + size.
 	// The store should be resized PRIOR to setting the memory
-	if offset+32 > uint64(len(m.store)) {
+	var empty = make([]byte, bit32)
+	if offset+bit32 > uint64(len(m.store)) {
 		panic("invalid memory: store empty")
 	}
 	// Zero the memory area
-	copy(m.store[offset:offset+32], []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0})
+	copy(m.store[offset:offset+bit32], empty)
```

Comment:
I don't see any reason why this would be faster. As it was, the compiler can replace the `[]byte{...}` with a constant, which it might not do with a dynamically allocated `empty`. 
I have a recollection that we added this ugly hack a long time ago, when we detected that it was indeed faster to initalize it like this.

Reply:
I think this performance gap is acceptable.